### PR TITLE
remove pair interaction

### PR DESCRIPTION
**Pull request by dorin-iancu**
- liquidity pool is no longer created, and users receive accepted token directly if they deposited launched tokens, and vice-versa
- extra rewards was removed
- changed unbond period from epochs to blocks (there was no longer any start epoch to choose)

### CODE REVIEW

File: dex/price-discovery/src/lib.rs [hunk 1]
```diff
@@ -17,15 +16,14 @@ const INVALID_PAYMENT_ERR_MSG: &[u8] = b"Invalid payment token";
 const BELOW_MIN_PRICE_ERR_MSG: &[u8] = b"Launched token below min price";
 pub const MIN_PRICE_PRECISION: u64 = 1_000_000_000_000_000_000;
 
-pub struct RewardsPair<M: ManagedTypeApi> {
-    pub lp_tokens_amount: BigUint<M>,
-    pub extra_rewards_amount: BigUint<M>,
+pub struct Rewards<M: ManagedTypeApi> {
```
Comment (claudiulataretu) [inline dex/price-discovery/src/lib.rs:19]: you can use `EsdtTokenPayment` instead of this new struct

Comment (dorin-iancu) [reply dex/price-discovery/src/lib.rs]: Done.

Comment (sasurobert) [inline dex/price-discovery/src/lib.rs]: rename to LockedTokens ? and you should have a unlock epoch - maybe next PR.

File: dex/price-discovery/src/lib.rs [hunk 2]
```diff
@@ -326,20 +254,30 @@ pub trait PriceDiscovery:
         &self,
         redeem_token_nonce: u64,
         redeem_token_amount: &BigUint,
-    ) -> RewardsPair<Self::Api> {
-        let total_lp_tokens = self.total_lp_tokens_received().get();
+    ) -> EsdtTokenPayment<Self::Api> {
```
Comment (sasurobert) [inline dex/price-discovery/src/lib.rs:257]: still not reward. but next PR can change some stuff.

**Review by sasurobert — APPROVED** [review]

File: dex/price-discovery/tests/tests_common.rs [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
 use elrond_wasm_debug::tx_mock::TxResult;
 use elrond_wasm_debug::{managed_biguint, testing_framework::*};
 use elrond_wasm_debug::{managed_token_id, rust_biguint, DebugApi};
 
-use price_discovery::create_pool::*;
 use price_discovery::redeem_token::*;
 use price_discovery::*;
 
-use pair_mock::*;
+// use pair_mock::*;
```
Comment (claudiulataretu) [inline dex/price-discovery/tests/tests_common.rs:11]: remove commented code

**Review by claudiulataretu — APPROVED** [review]

**Pull request merged**